includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -545,4 +332,91 @@ private function getHref( $hrefKey ) {
 	private function getSocialHrefs() {
 		return DesignSystemSharedLinks::getInstance()->getSocialHrefs( $this->lang );
 	}
+
+	private function getLinkTextObjectList( String $section, Array $linkKeys ) {
+		return array_map(
+			function ( $link ) use ( $section ) {
+				return $this->getLinkTextTranslatableObject(
+					'global-footer-' . $section . '-link-' . $link,
+					$link,
+					$section . '.' . $link
+				);
+			},
+			$linkKeys
+		);
+	}
+
+	private function getLinkTextTranslatableObject( $titleKey, $hrefKey, $trackingLabel ) {
```

Comment:
What I see here are methods that are only here to provide correct API structure. What do you think about creating separate classes (instead of arrays) to reflect API structure? Then instead of calling those methods you could simply instantiate class objects.